src/modulefinder/sentry_modulefinder_linux.c

```diff
@@ -162,8 +165,63 @@ get_code_id_from_elf(void *base, size_t *size_out)
     return NULL;
 }
 
-static sentry_value_t
-module_to_value(const sentry_module_t *module)
+static sentry_uuid_t
+get_code_id_from_text_fallback(void *base)
+{
+    const uint8_t *text = NULL;
+    size_t text_size = 0;
+
+    const uint8_t *addr = base;
+    // iterate over all the program headers, for 32/64 bit separately
+    const unsigned char *e_ident = addr;
+    if (e_ident[EI_CLASS] == ELFCLASS64) {
+        const Elf64_Ehdr *elf = base;
+        const Elf64_Shdr *strheader = (const Elf64_Shdr *)(addr + elf->e_shoff
+            + elf->e_shentsize * elf->e_shstrndx);
+
+        const char *names = (const char *)(addr + strheader->sh_offset);
+        for (int i = 0; i < elf->e_shnum; i++) {
+            const Elf64_Shdr *header = (const Elf64_Shdr *)(addr + elf->e_shoff
+                + elf->e_shentsize * i);
+            const char *name = names + 1 + header->sh_name;
+            if (header->sh_type == SHT_PROGBITS && strcmp(name, "text") == 0) {
+                text = addr + header->sh_offset;
+                text_size = header->sh_size;
+                break;
+            }
+        }
+    } else {
+        const Elf32_Ehdr *elf = base;
+        const Elf32_Shdr *strheader = (const Elf32_Shdr *)(addr + elf->e_shoff
+            + elf->e_shentsize * elf->e_shstrndx);
+
+        const char *names = (const char *)(addr + strheader->sh_offset);
+        for (int i = 0; i < elf->e_shnum; i++) {
+            const Elf32_Shdr *header = (const Elf32_Shdr *)(addr + elf->e_shoff
+                + elf->e_shentsize * i);
+            const char *name = names + 1 + header->sh_name;
+            if (header->sh_type == SHT_PROGBITS && strcmp(name, "text") == 0) {
+                text = addr + header->sh_offset;
+                text_size = header->sh_size;
+                break;
+            }
+        }
+    }
+
+    sentry_uuid_t uuid = sentry_uuid_nil();
+
+    // adapted from
+    // https://github.com/getsentry/symbolic/blob/8f9a01756e48dcbba2e42917a064f495d74058b7/debuginfo/src/elf.rs#L100-L110
+    size_t max = MAX(text_size, _SC_PAGESIZE);
```

Comment:
```suggestion
    size_t max = MIN(text_size, 4096);
```

- This needs to take the smaller of both, to ensure we read _at most_ 4k bytes.
- `_SC_PAGESIZE` is not guaranteed to be 4k, but Breakpad has 4k hardcoded.